src/analysis/special_functions/trigonometric.lean

```diff
@@ -1771,55 +1772,48 @@ end
 open filter
 open_locale topological_space
 
-lemma tendsto_sin_pi_div_two : tendsto sin (𝓝[set.Iio (π/2)] (π/2)) (𝓝 1) :=
-by { convert (@continuous.continuous_within_at _ _ _ _ _ (set.Iio (π/2)) (π/2) continuous_sin).tendsto,
-  simp }
+lemma tendsto_sin_pi_div_two : tendsto sin (𝓝[Iio (π/2)] (π/2)) (𝓝 1) :=
+begin
+  rw show (𝓝 1 = 𝓝 (sin (π/2))), by rw sin_pi_div_two,
+  exact continuous_sin.continuous_within_at
+end
 
-lemma tendsto_cos_pi_div_two : tendsto cos (𝓝[set.Iio (π/2)] (π/2)) (𝓝[set.Ioi 0] 0) :=
+lemma tendsto_cos_pi_div_two : tendsto cos (𝓝[Iio (π/2)] (π/2)) (𝓝[Ioi 0] 0) :=
 begin
-  change (tendsto cos (𝓝[set.Iio (π/2)] (π/2)) (_ ⊓ _)),
-  rw tendsto_inf,
-  split,
-  { convert (@continuous.continuous_within_at _ _ _ _ _ (set.Iio (π/2)) (π/2) continuous_cos).tendsto,
-    simp },
-  { rw tendsto_principal,
-    use set.Ioi (0:ℝ),
-    split,
-    { exact mem_nhds_sets is_open_Ioi (by simp [pi_div_two_pos]) },
-    { use set.Iio (π/2),
-      split,
-      { exact λ (x:ℝ), set.mem_Iio.mpr },
-      { intro x,
-        simp,
-        intros h1 h2,
-        exact cos_pos_of_mem_Ioo (by linarith) h2 } } },
+  apply tendsto_nhds_within_of_tendsto_nhds_of_eventually_within,
```

Comment:
I should point out this is the most important modification. The fact that `nhds_within` is defined as an `inf` is somehow an implementation detail. Here I'm using the public-facing interface to this concept.